server/channels/auth.js

```diff
@@ -48,13 +62,48 @@ const register = async (socket, data) => {
   emit(socket, AUTH_REGISTERED, { registered: true });
 };
 
+const createGenfors = async (socket, data) => {
+  const { password, title, date } = data;
+  if (verifyAdminPassword(password)) {
+    await addGenfors(title, date);
+    logger.info('Created genfors by administrative request.', { title });
+  } else {
+    logger.warn('Someone tried to authenticate as administrator.', { title });
+    emit(socket, 'AUTH_ERROR', { error: 'Ugyldig administratorpassord.' });
+  }
+};
+
 const listener = (socket) => {
   async function action(data) {
     switch (data.type) {
       case AUTH_REGISTER: {
         register(socket, data);
         break;
       }
+      case ADMIN_CREATE_GENFORS: {
+        createGenfors(socket, data);
+        break;
+      }
+      case ADMIN_LOGIN: {
+        const { password } = data;
+        if (verifyAdminPassword(password)) {
+          logger.info(`'${socket.request.user.name}' authenticated as admin using admin password.`);
+          // eslint-disable-next-line no-underscore-dangle
+          const updatedUser = await setUserPermissions(socket.request.user._id,
+            permissionLevel.IS_MANAGER);
+          emit(socket, AUTH_SIGNED_IN, {
+            username: updatedUser.username,
+            full_name: updatedUser.name,
+            logged_in: updatedUser.logged_in,
+            id: updatedUser._id, // eslint-disable-line no-underscore-dangle
+            permissions: updatedUser.permissions,
+          });
+        } else {
+          logger.info(`'${socket.request.user.name}' tried to authenticate as admin using admin password.`);
+          emit(socket, 'AUTH_ERROR', { error: 'Ugyldig administratorpassord.' });
```

Comment:
This code block would preferably be a function that is tested.

Reply:
I only see tests for `createGenfors`.
I never got as far as to test the actual listener since it's basically just a switch in the other channels.

Reply:
Oops, you mean `ADMIN_LOGIN` and not `ADMIN_CREATE_GENFORS`. I will fix.